File: bindings/go/ptiff/tile.go
package ptiff

// Tile is one image tile: its grid position and the raw byte payload.
//
// It mirrors Python's `ptiff.Tile` value object over the SWIG low-level
// surface, so the idiomatic object layer stays uniform across languages.
type Tile struct {
	Column   uint
	Row      uint
	Data     []byte
	ByteSize int64
}

// NewTile builds a Tile from a grid position and its payload. When byteSize is
// zero it is derived from the length of data.
func NewTile(column, row uint, data []byte, byteSize int64) Tile {
	if byteSize == 0 {
		byteSize = int64(len(data))
	}
	return Tile{Column: column, Row: row, Data: data, ByteSize: byteSize}
}

File: bindings/octave/mex/ptiff/Tile.m
classdef Tile
    %Tile  One image tile: grid position + raw byte payload (ptiff-octave).
    %
    %   Mirrors the SWIG binding's Tile value object: a small container with
    %   `column`, `row`, `data` (uint8 row vector) and `byte_size`.

    properties
        column
        row
        data
        byte_size
    end

    methods
        function obj = Tile(column, row, data, byte_size)
            obj.column = column;
            obj.row = row;
            obj.data = data;
            if nargin >= 4
                obj.byte_size = byte_size;
            else
                obj.byte_size = numel(data);
            end
        end
    end
end
